Bug 1274083, added Configuring Node Resources section

ahardin-rh opened the pull request:
https://bugzilla.redhat.com/show_bug.cgi?id=1274083

@adellape PTAL at this first draft

ahardin-rh commented:
@adellape, any first thoughts?

adellape commented on admin_guide/manage_nodes.adoc:
```diff
@@ -270,3 +270,26 @@ $ oadm manage-node <node1> <node2> \
 _<node2>_`), you can use the `--selector=_<node_selector>_` option to evacuate
 pods on selected nodes.
+
+== Configuring Node Resources
```
Add a unique anchor.

adellape commented on admin_guide/manage_nodes.adoc:
```diff
@@ -270,3 +270,26 @@ $ oadm manage-node <node1> <node2> \
 pods on selected nodes.
+
+== Configuring Node Resources
+Edit your node configuration file by passing kubelet arguments:
```
Maybe: "You can configure node resources by adding kubelet arguments to the <appropriate section of the node config>."

adellape commented on admin_guide/manage_nodes.adoc:
```diff
@@ -270,3 +270,26 @@ $ oadm manage-node <node1> <node2> \
+Edit your node configuration file by passing kubelet arguments:
+
+----
+$ vim /etc/origin/node/node-config
```
This needs root permissions, so switch $ to #.

adellape commented on admin_guide/manage_nodes.adoc:
```diff
@@ -270,3 +270,26 @@ $ oadm manage-node <node1> <node2> \
+
+Example options include:
+
+* `max-pods=40`: Number of Pods that can run on this Kubelet.
```
s/Pods/pod/

s/Kubelet/kubelet/

ahardin-rh commented:
@rjhowe completed my first pass at this. PTAL. Thanks!

adellape commented on admin_guide/manage_nodes.adoc:
```diff
@@ -270,3 +270,28 @@ $ oadm manage-node <node1> <node2> \
+[[configuring-node-resources]]
+== Configuring Node Resources
+You can configure node resources by adding kubelet arguments to the node
+configuration file:
```
Sorry my last comment on this spot got kinda messed up cuz I didn't wrap em in backticks. Basically we need to verify where in this file (e.g., what section/stanza, what parameter name) these should be set.

adellape commented on admin_guide/manage_nodes.adoc:
```diff
@@ -270,3 +270,28 @@ $ oadm manage-node <node1> <node2> \
+configuration file:
+
+----
+# vim /etc/origin/node/node-config
```
Needs `.yaml` at the end.

adellape commented on admin_guide/manage_nodes.adoc:
```diff
@@ -270,3 +270,28 @@ $ oadm manage-node <node1> <node2> \
+collection is always run. Default: 90%
+
+* `image-gc-low-threshold=80`: The percent of disk usage before which image garbage
+collection is never run. Lowest disk usage to garbage collect to. Default: 80%
```
This could possibly be a table, if you haven't tried it out as one already.

rjhowe commented:
@ahardin-rh  we really should show how this will look in the node-config.yaml 

Example:
```
....
kubeletArguments:
  max-pods
    - "40"
  resolv-conf
    - "/etc/resolv.conf"
  image-gc-high-threshold:
    - "90"
  image-gc-low-threshold:
    - "80"
...
```

Taking this even further we could reference how this would look in the ansible hosts file. 
```
openshift_node_kubelet_args={'max-pods': ['40'], 'resolv-conf': ['/etc/resolv.conf'],  'image-gc-high-threshold': ['90'], 'image-gc-low-threshold': ['80']}
```

ahardin-rh commented:
@rjhowe @adellape Thanks! Made some edits.

ahardin-rh commented:
@rjhowe @adellape Good to go?

adellape commented on admin_guide/manage_nodes.adoc:
```diff
@@ -270,3 +270,47 @@ $ oadm manage-node <node1> <node2> \
+[[configuring-node-resources]]
+== Configuring Node Resources
+You can configure node resources by adding kubelet arguments to the node
+configuration file (/etc/origin/node/node-config.yaml). Add the
```
Mark up the file name.

adellape commented on admin_guide/manage_nodes.adoc:
```diff
@@ -270,3 +270,47 @@ $ oadm manage-node <node1> <node2> \
+kubeletArguments:
+  max-pods <1>
+    - "40"
+  resolv-conf<2>
```
Space before the `<2>`.

adellape commented:
@ahardin-rh Two minor nits, otherwise LGTM.

ahardin-rh commented:
@adellape Fixed. Thanks!

ahardin-rh commented:
@rjhowe further edits?

ahardin-rh commented:
**revhistory note:**

(for Admin Guide --> Managing Nodes)

    Added the link:../admin_guide/manage_nodes.html#configuring-node-resources[Configuring Node Resources] section.

(for Install/Config --> Advanced Install)

    Added `*openshift_node_kubelet_args*` to the link:../install_config/install/advanced_install.html#configuring-ansible[Host Variables table].

adellape commented:
This had some follow-up, see https://github.com/openshift/openshift-docs/pull/1719 for new revhistory note.

adellape commented:
[rev_history]
|link:../admin_guide/manage_nodes.html[Managing Nodes]

|Added the link:../admin_guide/manage_nodes.html#configuring-node-resources[Configuring Node Resources] section.
%
